### DISCUSSION TITLE
How to keep the image centered when zoomed out?

### DISCUSSION THREAD

**Discussion by dertel** [openseadragon/openseadragon#2404] [Q&A]
I noticed that in one of the "in the wild" [examples](https://www.artsy.net/artwork/saint-clair-cemin-greece) from the OpenSeadragon website, the image stays centered at all times (horizontally and vertically) when the sides of the image are smaller than the viewport. And if you try to pan while zoomed out it "bounces" back to the middle.

How do I reproduce these two behaviors?

My current options in case that helps:

```
{
  element: viewerEl,
  tileSources,
  zoomInButton,
  zoomOutButton,
  showHomeControl: false,
  showFullPageControl: false,
  showSequenceControl: false,
  animationTime: 0.5,
  constrainDuringPan: true,
  visibilityRatio: 1,
  minZoomImageRatio: 1,
}
```

**Comment by iangilman** [comment]
I believe `visibilityRatio: 1` is what they used to keep it in the center, but I believe the behavior of that feature has changed in the more recent versions. Now it just keeps it entirely on the screen, but doesn't center it, right? This is because that centering behavior was really just a side effect, and was causing problems with other scenarios.

It wouldn't be too difficult for you to write your own centering behavior if you want… Watch for `canvas-drag-end` events, check to see if you're zoomed out enough, and do a `viewer.viewport.panTo` to get it back in the center.

If you want the "bouncing" behavior (rather than it limiting it while you pan), you should set `constrainDuringPan: false`.

**Comment by dertel** [comment]
> Now it just keeps it entirely on the screen, but doesn't center it, right?

Yes exactly.

Thanks for the tip on implementing the centering behavior. I was able to use this technique to reproduce part of the centering effect from the 2.x . There are some other centering behaviors of the 2.x version that this doesn't cover however.

For example when you zoom in with the scroll wheel in the [example](https://www.artsy.net/artwork/saint-clair-cemin-greece) you zoom towards the mouse point — this works the same in OpenSeadragon 2.x and 4.x. But when you zoom out, in 4.x it zooms out centered around the mouse point again, causing the image to be off-center. In 2.x, it gradually moves towards the center of the viewport.

Feel free to close this thread as it seems that the move away from automatically centering the image was intentional in more recent versions. But if you think you have a tip on the scroll wheel scenario mentioned above I would be grateful for your help. It would be great to be able to upgrade to the most recent version of OSD.

Thanks for the quick reply!

**Reply by iangilman** [reply]
This reply isn't so quick! I've been away on vacation...

Are you still interested in implementing the centering on scroll? I think it would be similar to the pan version, but instead you watch for `zoom` events and do some panning as needed in response to those. Those events fire once per zoom, not once per animation frame, so if you want the effect to happen continuously during the animation, you should watch for `animation` events instead.
